core/test/config/preconditioner.cpp

```diff
@@ -297,6 +299,101 @@ struct Jacobi
 };
 
 
+struct Sor
+    : PreconditionerConfigTest<::gko::preconditioner::Sor<float, gko::int32>,
+                               ::gko::preconditioner::Sor<double, gko::int32>> {
+    using Ir = gko::solver::Ir<float>;
+
+    static pnode::map_type setup_base()
+    {
+        return {{"type", pnode{"preconditioner::Sor"}}};
+    }
+
+    static void change_template(pnode::map_type& config_map)
+    {
+        config_map["value_type"] = pnode{"float32"};
+    }
+
+    template <bool from_reg, typename ParamType>
```

Comment:
I decided against using that. There is nothing new tested in that case. The parsing for SOR is already checked, and getting the right objects from the registry is also tested in a bunch of other tests. So it would only add noise to the test.

Reply:
The others' test can not ensure that this can handle them in the same way.
We do not have any thing such as the class hierarchy to ensure it uses the function, which can handle the generated object and config, from the test.

Reply:
I don't think there is another way to get a factory from a config other than `parse_or_get_factory`. For this function, we have already enough tests to check that the factory is either parsed, or extracted from the registry. As long as we only keep this one parsing function, there is no need to add more tests.

Reply:
Yes, I know we use the function and should be correct when we use the function.
My question is how the test can know we indeed use the function.
If the test check both behaviors, it does not check what the function is but ensure both behaviors match expectation.

Reply:
You're right, I was just a bit too lazy to also add those test.